core/workflows/workflow_base.py

```
@@ -0,0 +1,196 @@
+        """
+        if not self.config.checkpoint_enabled:
+            return
+
+        checkpoint_path = self.config.staging_path / f"{self.config.name}_checkpoint.json"
+
+        try:
+            import json
+            with open(checkpoint_path, 'w') as f:
+                json.dump(checkpoint_data, f, default=str)
+            logger.debug(f"Checkpoint saved to {checkpoint_path}")
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)